app/models/foreman_docker/service/actions/container/pull.rb

```diff
@@ -0,0 +1,28 @@
+module ForemanDocker
+  module Service
+    module Actions
+      module Container
+        class Pull < ::Actions::EntryAction
+          def plan(container)
+            action_subject(container)
+            plan_self(:container => container.id)
+          end
+
+          def run
+            container = ::Container.find(input[:container])
+            container.compute_resource.create_image(:fromImage => container.repository_pull_url)
```

Comment:
@eLobato in same order:

- `:done` is required so the action finishes after the `create_image` is done not sooner to be able to report any errors in the action.
- yes `event` is same object as the object pushed to `suspended_action` with `<<`
- yes, on first `run` the event is `nil`, so actually this is the only one limitation - not sending nil `event`s